Yoast/javascript
# Fix parsing of (unintended) comments

## Summary
This PR can be summarized in the following changelog entry:
* _[not-user-facing]_ Parsing comments inside paragraphs and headings does not break the analysis anymore.

Apparently, the browser parses incomplete tags as comments, like this:
```html
<p>This is a paragraph </ with an incomplete closing tag.</p>
```
to this:
```html
<p>This is a paragraph </p>
```
This broke our tree parser, since it could not handle comments inside paragraphs or headings.

This PR fixes this issue by returning the formatting elements to parse5 when it tries to retrieve the comment node from the tree.

## Relevant technical choices
* We refactored the code to allow `Ignored` nodes as formatting in a heading or paragraph. Previously, we transformed these nodes to `FormattingElement`s first.
  * We chose this option to be able to keep track of the source code location and content.

## Testing instructions
* In one of the example texts, add `</` (be mindful of PHPStorm's autocomplete) in a heading and a paragraph, and see that it still works. Do the same with ``, and `<script>blabla</script>`. Verify that this breaks on develop but not on this branch.

Fixes #38
